pkgs/development/libraries/java/saxon/default.nix

```
@@ -36,21 +37,35 @@ let
       {
         inherit pname version src;
 
-        nativeBuildInputs = [ unzip ];
-
-        buildCommand = ''
-          unzip $src -d $out
-          mkdir -p $out/bin $out/share $out/share/java
-          cp -s "$out"/*.jar "$out/share/java/"  # */
-          rm -rf $out/notices
-          mv $out/doc $out/share
-          cat > $out/bin/${mainProgram} <<EOF
-          #! $shell
-          export JAVA_HOME=${jre}
-          exec ${jre}/bin/java -jar $out/${jar'}.jar "\$@"
-          EOF
-          chmod a+x $out/bin/${mainProgram}
-        '';
+        nativeBuildInputs = [
+          unzip
+          makeWrapper
+        ];
+
+        unpackCmd = "unzip $src -d src";
```

The same effect could also be achieved with this:
```suggestion
        sourceRoot = ".";
```
Up to you to choose whichever you prefer.

---

Great suggestion, thanks, @rvl! And thanks for the quick follow-up, review and approval, much appreciated 👍